src/api.js

```diff
@@ -464,56 +511,104 @@ var remoteAuthentication = function(options, entry) {
           return false;
         }
 
-        // If we're not given an array, we assume it's a set of parameters that
-        // must be used to parameterize the original scopesets
-        if (!(scopesets instanceof Array)) {
-          var params = scopesets;
-          scopesets = _.cloneDeepWith(entry.scopes, function(scope) {
+        req.hasAuthed = false;
+        let missing = [];
+        let hasConditional = false;
+
+        var _splatParams = (scope) => scope.replace(/<([^>]+)>/g, function(match, param) {
+          var value = _.at(params, param)[0];
+          if (value !== undefined) {
+            return value;
+          }
+          missing.push(match); // If any are left undefined, we can't be done yet
+          return match;
+        });
+
+        var _expandExpressionTemplate = (template) => {
+          var key = Object.keys(template)[0];
+          var subexpressions = [];
+          template[key].forEach(scope => {
             if (typeof scope === 'string') {
-              return scope.replace(/<([^>]+)>/g, function(match, param) {
-                var value = params[param];
-                return value === undefined ? match : value;
+              subexpressions.push(_splatParams(scope));
+            } else if (_.isObject(scope) && scope.for && scope.in && scope.each) {
+              let subs = _.at(params, scope.in)[0];
+              if (!subs) {
+                missing.push(scope.in);
+                subexpressions.push(scope);
+                return;
+              }
+              subs.forEach(param => {
+                subexpressions.push(_splatParams(scope.each.replace(`<${scope.for}>`, param)));
```

Comment:
So, answering my earlier question, nesting is not allowed.  That's fine, but should be documented.